contrib/mesos/pkg/executor/executor.go

```diff
@@ -162,24 +145,48 @@ func New(config Config) *Executor {
 		suicideWatch:         &suicideTimer{},
 		shutdownAlert:        config.ShutdownAlert,
 		exitFunc:             config.ExitFunc,
-		podStatusFunc:        config.PodStatusFunc,
 		staticPodsConfigPath: config.StaticPodsConfigPath,
 		launchGracePeriod:    launchGracePeriod,
 		nodeInfos:            config.NodeInfos,
 		initCompleted:        make(chan struct{}),
+		registry:             config.Registry,
 	}
+
 	runtime.On(k.initCompleted, k.runSendLoop)
 
-	po := newPodObserver(config.PodLW, k.updateTask, k.terminate)
-	runtime.On(k.initCompleted, po.run)
+	k.podWatcher = newPodWatcher(k.registry.watch())
+	runtime.On(k.initCompleted, k.podWatcher.run)
 
 	return k
 }
 
+// Done returns a chan that closes when the executor is shutting down
+func (k *Executor) Done() <-chan struct{} {
+	return k.terminate
+}
+
 func (k *Executor) Init(driver bindings.ExecutorDriver) {
 	defer close(k.initCompleted)
+
 	k.killKubeletContainers()
 	k.resetSuicideWatch(driver)
+
+	// an active pod-task was deleted, alert mesos
+	k.podWatcher.filter(func(pod *RegisteredPod) bool {
+		switch pod.event {
+		case RegistrationEventIncompatibleUpdate:
```

Comment:
should probably `return false` for this case; the pod will be deleted and we'll get another event for that later